UI/window-basic-main.cpp

```diff
@@ -5170,18 +5172,11 @@ void OBSBasic::ClearSceneData()
 
 void OBSBasic::closeEvent(QCloseEvent *event)
 {
-	if (!startStreamingFuture.future.isFinished() &&
-	    !startStreamingFuture.future.isCanceled()) {
-		startStreamingFuture.future.onCanceled(
-			this, [basic = QPointer{this}] {
-				if (basic)
-					basic->close();
-			});
-		startStreamingFuture.cancelAll();
-		event->ignore();
-		return;
-	} else if (startStreamingFuture.future.isCanceled() &&
-		   !startStreamingFuture.future.isFinished()) {
+	/* Wait for multitrack video stream to start/finish processing in the background */
+	if (setupStreamingGuard.valid() &&
+	    setupStreamingGuard.wait_for(std::chrono::seconds{0}) !=
+		    std::future_status::ready) {
+		QTimer::singleShot(1000, this, &OBSBasic::close);
```

Comment:
`OBSBasic::close` causes another call to `closeEvent`, so this check gets repeated until `setupStreamingGuard` is actually resolved, so the behavior should be fairly similar to the previous code (except the previous code scheduled a continuation that fired immediately after the future was resolved, as opposed to this)

an alternative would be to spawn a "task" on another thread that waits for `setupStreamingGuard` to resolve and then immediately `QMetaObject::invoke`s `OBSBasic::close`; keeping track of that task is somewhat annoying though in that it requires yet another `std::future` or some other synchronization primitive

this could then change to something like
```c++
if (setupStreamingGuard.valid() && setupStreamingGuard.wait_for(std::chrono::seconds{0}) != std::future_status::ready) {
   if (!setupStreamingShutdownMonitor.valid()) {
      auto monitor = std::make_shared<StartMultitrackVideoStreamingGuard>();
      setupStreamingShutdownMonitor = monitor.GetFuture();
      QThreadPool::globalInstance()->start([self = this, setupStreamingGuard, monitor] {
        setupStreamingGuard.wait();
        monitor.reset();
        QMetaObject::invokeMethod(self, &OBSBasic::close);
      });
    }
}
if (setupStreamingShutdownMonitor.valid()) {
  event->ignore();
  return;
}
```

ultimately it all depends on `setupStreamingGuard` only being resolved when the background task is finished (same as `startStreamingFuture` previously, except `setupStreamingGuard` requires more manual work to keep alive long enough unfortunately

Reply:
It's probably fine as-is. I just wanted to note it.